EIPS/eip-7867.md

```diff
@@ -0,0 +1,553 @@
+
+## Motivation
+
+While the base EIP-5792 specification works extremely well for smart contract wallets, it does not allow the expression of the full range of flow control options that wallets can implement. For example, a dapp may only be submitting a batch for gas savings and not care about whether all calls are reverted on failure. A wallet may only be able to offer a limited form of atomicity through block builder backchannels, but that may be sufficient for a trading platform.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### RPC Interface
+
+The following subsections are modifications to the API endpoints from EIP-5792.
+
+If a request does not match the schema defined below, the wallet MUST reject the request with an error code of 0 <!-- TODO -->.
+
+#### `wallet_sendCalls`
+
+The following JSON Schema SHALL be inserted, in the request object, as values of either the batch-scope or call-scope `capabilities` objects (as appropriate) with a key of `flow-control`.
+
+##### Batch-scope
+
+###### Schema
+
+```json
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "type": "object",
+    "additionalProperties": false,
+    "properties": {
+      "optional": {
+        "type": "boolean"
+      },
+      "atomicity": {
+        "enum": ["strict", "loose", "none"]
+      }
+    }
+}
+```
+
+###### Example Request
+
+```js
+[
+  {
+    "version": "1.0",
+    "from": "0xd46e8dd67c5d32be8058bb8eb970870f07244567",
+    "chainId": "0x01",
+    "calls": [],
+    "capabilities": {
+      "flow-control": {
+        "atomicity": "loose"
+      }
+    }
+  }
+]
+
+```
+
+
+##### Call-scope
+
+###### Schema
+
+```json
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "type": "object",
+    "additionalProperties": false,
+    "properties": {
+      "optional": {
+        "type": "boolean"
+      },
+      "on-failure": {
+        "enum": ["rollback", "halt", "continue"]
+      }
+    }
+}
+```
+
+###### Example Request
+
+```js
+[
+  {
+    "version": "1.0",
+    "from": "0xd46e8dd67c5d32be8058bb8eb970870f07244567",
+    "chainId": "0x01",
+    "calls": [
+      {
+        "to": "0xd46e8dd67c5d32be8058bb8eb970870f07244567",
+        "value": "0x182183",
+        "data": "0xfbadbaf01",
+        "capabilities": {
+            "flow-control": {
+                "on-failure": "continue"
```

Comment:
what exactly is a wallet supposed to do in this case when the flow fails? what does continue mean? like to continue with the execution of the next calls? what if there's only one call in the array?

Reply:
Hopefully that's defined well enough later in the document (and let me know if it isn't), but a failed call with `"on-failure": "continue"` is just slipped, and the next call is executed.